Yoast/YoastSEO.js
# Adds a filter to the passive analysis for stopwords and stop characters without preceding space

## Summary

This PR can be summarized in the following changelog entry: Adds a filter to avoid incorrectly marking participles as passive when there is a stopword or relevant puncuation mark between the auxiliary and the passive that isn't preceded by a space. For example, "It's a hands-free, voice-controlled device" will not be marked as passive because there is a comma between the auxiliary "it's" and the -ed word "voice-controlled". Also adds some additional stopwords (“–“, “and”, “or”, “but”) and non-verbs ending in -ed exceptions ("based”, “located”, “red”, “related”).

## Test instructions

This PR can be tested by following these steps:

- Use the browserified example. Don't forget to run `grunt build:js`.
- Paste a passive English sentence (e.g., `I was helped`)
- Add a string including a stop word (e.g., `'ll` and `'ve` such as in `she'll` and `we've`) or relevant punctuation marks (`:` and `,`) between `was` and `helped`.
- To test the new stopwords (surrounded by spaces), add one of these words between `was` and `helped`.
- To test the new non-verbs ending in -ed exceptions, replace `helped` with one of these words.
- Make sure none of these sentences are marked as passive


Fixes #1223
Fixes #962
